packages/components/src/internal/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -175,91 +172,41 @@ class SampleTypeDesignerImpl extends React.PureComponent<Props & InjectedBaseDom
     }
 
     componentDidMount = async (): Promise<void> => {
-        const { includeDataClasses, setSubmitting } = this.props;
+        const { includeDataClasses, setSubmitting, isValidParentOptionFn } = this.props;
         const { model } = this.state;
 
         try {
-            const results = await initSampleSetSelects(!model.isNew(), includeDataClasses, model.containerPath);
-            this.initParentOptions(results);
+            const { parentOptions, parentAliases } = await initParentOptionsSelects(
+                true,
+                includeDataClasses,
+                model.containerPath,
+                isValidParentOptionFn,
+                model.isNew() ? NEW_SAMPLE_SET_OPTION : null,
+                model.importAliases,
+                'sampleset-parent-import-alias-',
+                this.formatLabel
+            );
+            this.setState({
+                model: model.merge({ parentAliases }) as SampleTypeModel,
+                parentOptions,
+            });
         } catch (error) {
             setSubmitting(false, () => {
                 this.setState({ error: resolveErrorMessage(error) });
             });
         }
     };
 
-    formatLabel = (name: string, prefix: string, containerPath?: string): string => {
+    formatLabel = (name: string, prefix: string, isDataClass?: boolean, containerPath?: string): string => {
         const { includeDataClasses, useSeparateDataClassesAliasMenu, showParentLabelPrefix } = this.props;
+        const { model } = this.state;
+        if (name === model?.name && !isDataClass) return NEW_SAMPLE_SET_OPTION.label;
+
         return includeDataClasses && !useSeparateDataClassesAliasMenu && showParentLabelPrefix
             ? `${prefix}: ${name} (${containerPath})`
             : name;
     };
 
-    initParentOptions = (responses: ISelectRowsResult[]): void => {
```

Comment:
glad you were able to refactor this all into `initParentOptionsSelects` so that it could be generalized/reused

Reply:
Yes, agreed. If you want to take it a step further this can be updated to be `async` and use `selectRows`.

```ts
export async function initParentOptionsSelects(
    includeSampleTypes: boolean,
    includeDataClasses: boolean,
    containerPath: string,
    isValidParentOptionFn?: (row: any, isDataClass: boolean) => boolean,
    newTypeOption?: any,
    importAliases?: Map<string, string>,
    idPrefix?: string,
    formatLabel?: (name: string, prefix: string, isDataClass?: boolean, containerPath?: string) => string
): Promise<{
    parentAliases: Map<string, IParentAlias>;
    parentOptions: IParentOption[];
}> {
    const promises: Array<Promise<SelectRowsResponse>> = [];

    // Get Sample Types
    if (includeSampleTypes) {
        promises.push(
            selectRows({
                containerPath,
                schemaQuery: SCHEMAS.EXP_TABLES.SAMPLE_SETS,
                columns: 'LSID, Name, RowId, Folder',
                containerFilter: Query.containerFilter.currentPlusProjectAndShared,
            })
        );
    }

    // Get Data Classes
    if (includeDataClasses) {
        promises.push(
            selectRows({
                containerPath,
                schemaQuery: SCHEMAS.EXP_TABLES.DATA_CLASSES,
                columns: 'LSID, Name, RowId, Folder, Category',
                containerFilter: Query.containerFilter.currentPlusProjectAndShared,
            })
        );
    }

    const responses = await Promise.all(promises);

    const sets: IParentOption[] = [];
    responses.forEach(result => {
        const isDataClass = result.schemaQuery === SCHEMAS.EXP_TABLES.DATA_CLASSES;
        const prefix = isDataClass ? DATA_CLASS_IMPORT_PREFIX : SAMPLE_SET_IMPORT_PREFIX;
        const labelPrefix = isDataClass ? 'Data Class' : 'Sample Type';

        result.rows.forEach(row => {
            if (isValidParentOptionFn) {
                if (!isValidParentOptionFn(row, isDataClass)) return;
            }

            const name = caseInsensitive(row, 'Name')?.value;
            const path = caseInsensitive(row, 'Folder')?.displayValue;

            sets.push({
                value: prefix + name,
                label: formatLabel ? formatLabel(name, labelPrefix, isDataClass, path) : name,
                schema: isDataClass ? SCHEMAS.DATA_CLASSES.SCHEMA : SCHEMAS.SAMPLE_SETS.SCHEMA,
                query: name, // Issue 33653: query name is case-sensitive for some data inputs (sample parents)
            });
        });

        if (newTypeOption) {
            if (
                (!isDataClass && newTypeOption.schema === SCHEMAS.SAMPLE_SETS.SCHEMA) ||
                (isDataClass && newTypeOption.schema !== SCHEMAS.SAMPLE_SETS.SCHEMA)
            )
                sets.push(newTypeOption);
        }
    });

    const parentOptions = sets.sort(naturalSortByProperty('label'));

    let parentAliases = Map<string, IParentAlias>();

    if (importAliases) {
        const initialAlias = Map<string, string>(importAliases);
        initialAlias.forEach((val, key) => {
            const newId = generateId(idPrefix);
            const parentValue = parentOptions.find(opt => opt.value === val);
            if (!parentValue) {
                // parent option might have been filtered out by isValidParentOptionFn
                return;
            }

            parentAliases = parentAliases.set(newId, {
                id: newId,
                alias: key,
                parentValue,
                ignoreAliasError: false,
                ignoreSelectError: false,
            } as IParentAlias);
        });
    }

    return { parentAliases, parentOptions };
}
```

with the corollary change in LKSM to the `isValidParentOptionFn`:

```ts
import { DATA_CLASS_CATEGORY_SOURCE } from '../SourceTypes/constants';
import { caseInsensitive } from '@labkey/components';

export function isValidParentOption(row: any, isDataClass: boolean): boolean {
    if (!isDataClass) return true;
    return DATA_CLASS_CATEGORY_SOURCE === caseInsensitive(row, 'Category')?.value;
}
```